quarkus-debezium-parent/quarkus-debezium-engine-parent/runtime/src/main/java/io/quarkus/debezium/engine/DefaultQualifiedTableNameResolver.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.quarkus.debezium.engine;
+
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.source.SourceRecord;
+
+import io.debezium.engine.RecordChangeEvent;
+
+class DefaultQualifiedTableNameResolver implements FullyQualifiedTableNameResolver {
+
+    @Override
+    public String resolve(RecordChangeEvent<SourceRecord> event) {
+        SourceRecord record = event.record();
+        Struct payload = (Struct) record.value();
+        String table = ((Struct) payload.get("source")).getString("table");
```

Comment:
No need for typecasting, this and the rest of the code can work like
```suggestion
        String table = payload.getStruct("source").getString("table");
```